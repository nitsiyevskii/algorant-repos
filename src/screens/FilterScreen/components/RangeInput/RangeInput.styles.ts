import { createStyles } from 'src/utils/createStyles';

export const useStyles = createStyles((theme) => ({
  container: {
    marginBottom: 8,
  },
  label: {
    fontSize: theme.fonts.size.lg,
    fontWeight: theme.fonts.weight.semibold,
    color: theme.colors.text.primary,
    marginBottom: 16,
  },
  rangeContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  inputWrapper: {
    flex: 1,
  },
  inputLabel: {
    fontSize: theme.fonts.size.sm,
    color: theme.colors.text.secondary,
    marginBottom: 8,
  },
  separator: {
    fontSize: theme.fonts.size.lg,
    color: theme.colors.text.secondary,
    marginHorizontal: 12,
    marginTop: 20,
  },
  input: {
    backgroundColor: theme.colors.input.background,
    borderWidth: 1,
    borderColor: theme.colors.input.border,
    borderRadius: 8,
    paddingHorizontal: 12,
    paddingVertical: 10,
    fontSize: theme.fonts.size.md,
    color: theme.colors.text.primary,
  },
}));

