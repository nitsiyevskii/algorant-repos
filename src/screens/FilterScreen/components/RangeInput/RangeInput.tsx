import React from "react";
import { View, Text, TextInput } from "react-native";
import { useTheme } from "src/hooks/useTheme";
import { parsePositiveInteger } from "src/utils/number";
import { useStyles } from "./RangeInput.styles";

interface RangeInputProps {
  label: string;
  minValue: number | null;
  maxValue: number | null;
  onMinChange: (value: number | null) => void;
  onMaxChange: (value: number | null) => void;
  minPlaceholder?: string;
  maxPlaceholder?: string;
}

export const RangeInput: React.FC<RangeInputProps> = ({
  label,
  minValue,
  maxValue,
  onMinChange,
  onMaxChange,
  minPlaceholder = "0",
  maxPlaceholder = "âˆž",
}) => {
  const theme = useTheme();
  const styles = useStyles();

  const handleMinChange = (text: string) => {
    const value = parsePositiveInteger(text);
    onMinChange(value);
  };

  const handleMaxChange = (text: string) => {
    const value = parsePositiveInteger(text);
    onMaxChange(value);
  };

  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      <View style={styles.rangeContainer}>
        <View style={styles.inputWrapper}>
          <Text style={styles.inputLabel}>Min</Text>
          <TextInput
            style={styles.input}
            value={minValue?.toString() || ""}
            onChangeText={handleMinChange}
            placeholder={minPlaceholder}
            placeholderTextColor={theme.colors.input.placeholder}
            keyboardType="number-pad"
            testID={`range-input-min-${label.toLowerCase()}`}
          />
        </View>
        <Text style={styles.separator}>-</Text>
        <View style={styles.inputWrapper}>
          <Text style={styles.inputLabel}>Max</Text>
          <TextInput
            style={styles.input}
            value={maxValue?.toString() || ""}
            onChangeText={handleMaxChange}
            placeholder={maxPlaceholder}
            placeholderTextColor={theme.colors.input.placeholder}
            keyboardType="number-pad"
            testID={`range-input-max-${label.toLowerCase()}`}
          />
        </View>
      </View>
    </View>
  );
};
