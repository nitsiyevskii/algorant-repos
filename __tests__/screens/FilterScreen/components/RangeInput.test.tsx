import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { RangeInput } from 'src/screens/FilterScreen/components/RangeInput/RangeInput';

describe('RangeInput', () => {
  const mockOnMinChange = jest.fn();
  const mockOnMaxChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with label', () => {
    const { getByText } = render(
      <RangeInput
        label="Stars"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    expect(getByText('Stars')).toBeTruthy();
    expect(getByText('Min')).toBeTruthy();
    expect(getByText('Max')).toBeTruthy();
  });

  it('displays current min and max values', () => {
    const { getByDisplayValue } = render(
      <RangeInput
        label="Forks"
        minValue={10}
        maxValue={100}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    expect(getByDisplayValue('10')).toBeTruthy();
    expect(getByDisplayValue('100')).toBeTruthy();
  });

  it('displays empty inputs when values are null', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Stars"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const minInput = getByTestId('range-input-min-stars');
    const maxInput = getByTestId('range-input-max-stars');

    expect(minInput.props.value).toBe('');
    expect(maxInput.props.value).toBe('');
  });

  it('calls onMinChange with valid positive integer', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Stars"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const minInput = getByTestId('range-input-min-stars');
    fireEvent.changeText(minInput, '50');

    expect(mockOnMinChange).toHaveBeenCalledWith(50);
  });

  it('calls onMaxChange with valid positive integer', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Forks"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const maxInput = getByTestId('range-input-max-forks');
    fireEvent.changeText(maxInput, '200');

    expect(mockOnMaxChange).toHaveBeenCalledWith(200);
  });

  it('calls onMinChange with null for empty string', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Stars"
        minValue={10}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const minInput = getByTestId('range-input-min-stars');
    fireEvent.changeText(minInput, '');

    expect(mockOnMinChange).toHaveBeenCalledWith(null);
  });

  it('calls onMinChange with null for invalid input', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Stars"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const minInput = getByTestId('range-input-min-stars');
    fireEvent.changeText(minInput, 'abc');

    expect(mockOnMinChange).toHaveBeenCalledWith(null);
  });

  it('calls onMinChange with null for negative numbers', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Stars"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const minInput = getByTestId('range-input-min-stars');
    fireEvent.changeText(minInput, '-5');

    expect(mockOnMinChange).toHaveBeenCalledWith(null);
  });

  it('calls onMinChange with null for zero', () => {
    const { getByTestId } = render(
      <RangeInput
        label="Stars"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    const minInput = getByTestId('range-input-min-stars');
    fireEvent.changeText(minInput, '0');

    expect(mockOnMinChange).toHaveBeenCalledWith(null);
  });

  it('uses custom placeholders', () => {
    const { getByPlaceholderText } = render(
      <RangeInput
        label="Custom"
        minValue={null}
        maxValue={null}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
        minPlaceholder="Min value"
        maxPlaceholder="Max value"
      />
    );

    expect(getByPlaceholderText('Min value')).toBeTruthy();
    expect(getByPlaceholderText('Max value')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <RangeInput
        label="Stars"
        minValue={10}
        maxValue={100}
        onMinChange={mockOnMinChange}
        onMaxChange={mockOnMaxChange}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});

