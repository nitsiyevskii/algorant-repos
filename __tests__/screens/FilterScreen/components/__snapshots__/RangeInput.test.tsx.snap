// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput matches snapshot 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 16,
      }
    }
  >
    Stars
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#3C3C43",
            "fontSize": 14,
            "marginBottom": 8,
          }
        }
      >
        Min
      </Text>
      <TextInput
        keyboardType="number-pad"
        onChangeText={[Function]}
        placeholder="0"
        placeholderTextColor="#8E8E93"
        style={
          {
            "backgroundColor": "#F2F2F7",
            "borderColor": "#C7C7CC",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "#000000",
            "fontSize": 16,
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
        testID="range-input-min-stars"
        value="10"
      />
    </View>
    <Text
      style={
        {
          "color": "#3C3C43",
          "fontSize": 18,
          "marginHorizontal": 12,
          "marginTop": 20,
        }
      }
    >
      -
    </Text>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#3C3C43",
            "fontSize": 14,
            "marginBottom": 8,
          }
        }
      >
        Max
      </Text>
      <TextInput
        keyboardType="number-pad"
        onChangeText={[Function]}
        placeholder="âˆž"
        placeholderTextColor="#8E8E93"
        style={
          {
            "backgroundColor": "#F2F2F7",
            "borderColor": "#C7C7CC",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "#000000",
            "fontSize": 16,
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
        testID="range-input-max-stars"
        value="100"
      />
    </View>
  </View>
</View>
`;
